aws_lambda_powertools/event_handler/api_gateway.py

```diff
@@ -596,14 +603,14 @@ def _path_starts_with(path: str, prefix: str):
 
     def _not_found(self, method: str) -> ResponseBuilder:
         """Called when no matching route was found and includes support for the cors preflight response"""
-        headers = {}
+        headers: Dict[str, List[str]] = defaultdict(list)
         if self._cors:
             logger.debug("CORS is enabled, updating headers.")
             headers.update(self._cors.to_dict())
 
             if method == "OPTIONS":
                 logger.debug("Pre-flight request detected. Returning CORS with null response")
-                headers["Access-Control-Allow-Methods"] = ",".join(sorted(self._cors_methods))
+                headers["Access-Control-Allow-Methods"].append(",".join(sorted(self._cors_methods)))
```

Comment:
np: do we need to sort it? previous iteration I know. It could save some cycles if unnecessary.